pkg/sdkserver/localsdk.go

```diff
@@ -384,28 +387,139 @@ func (l *LocalSDKServer) stopReserveTimer() {
 // [Stage:Alpha]
 // [FeatureFlag:PlayerTesting]
 func (l *LocalSDKServer) PlayerConnect(ctx context.Context, id *alpha.PlayerID) (*alpha.Bool, error) {
-	panic("implement me")
+	if !runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		return nil, errors.New(string(runtime.FeaturePlayerTracking) + " not enabled")
+	}
+	logrus.WithField("playerID", id.PlayerID).Info("Player Connected")
+	l.gsMutex.Lock()
+	defer l.gsMutex.Unlock()
+
+	if l.gs.Status.Players == nil {
+		l.gs.Status.Players = &sdk.GameServer_Status_PlayerStatus{}
+	}
+
+	// the player is already connected, return false.
+	for _, playerID := range l.gs.Status.Players.IDs {
+		if playerID == id.PlayerID {
+			return &alpha.Bool{Bool: false}, nil
+		}
+	}
+
+	if l.gs.Status.Players.Count >= l.gs.Status.Players.Capacity {
+		return &alpha.Bool{}, errors.New("Players are already at capacity")
+	}
+
+	l.gs.Status.Players.IDs = append(l.gs.Status.Players.IDs, id.PlayerID)
+	l.gs.Status.Players.Count = int64(len(l.gs.Status.Players.IDs))
+
+	l.update <- struct{}{}
+	l.recordRequestWithValue("playerconnect", "1234", "PlayerIDs")
+	return &alpha.Bool{Bool: true}, nil
 }
 
 // PlayerDisconnect should be called when a player disconnects.
 // [Stage:Alpha]
 // [FeatureFlag:PlayerTesting]
 func (l *LocalSDKServer) PlayerDisconnect(ctx context.Context, id *alpha.PlayerID) (*alpha.Bool, error) {
-	panic("implement me")
+	if !runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		return nil, errors.New(string(runtime.FeaturePlayerTracking) + " not enabled")
+	}
+	logrus.WithField("playerID", id.PlayerID).Info("Player Disconnected")
+	l.gsMutex.Lock()
+	defer l.gsMutex.Unlock()
+
+	if l.gs.Status.Players == nil {
+		l.gs.Status.Players = &sdk.GameServer_Status_PlayerStatus{}
+	}
+
+	found := -1
+	for i, playerID := range l.gs.Status.Players.IDs {
+		if playerID == id.PlayerID {
+			found = i
+			break
+		}
+	}
+	if found == -1 {
+		return &alpha.Bool{Bool: false}, nil
+	}
+
+	l.gs.Status.Players.IDs = append(l.gs.Status.Players.IDs[:found], l.gs.Status.Players.IDs[found+1:]...)
+	l.gs.Status.Players.Count = int64(len(l.gs.Status.Players.IDs))
+
+	l.update <- struct{}{}
+	l.recordRequestWithValue("playerdisconnect", "", "PlayerIDs")
+	return &alpha.Bool{Bool: true}, nil
+}
+
+// IsPlayerConnected returns if the playerID is currently connected to the GameServer.
+// [Stage:Alpha]
+// [FeatureFlag:PlayerTesting]
```

Comment:
Should here be `PlayerTracking` as in FeatureFlag?
```suggestion
// [FeatureFlag:PlayerTracking]
```

Reply:
Thank you - I missed that.